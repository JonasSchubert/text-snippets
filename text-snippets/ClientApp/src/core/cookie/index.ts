export const appName = "dotnetcore-vue-template";

export const daysTilExpired: { [key: string]: number } = {};

export const keys: { [key: string]: string } = {};
