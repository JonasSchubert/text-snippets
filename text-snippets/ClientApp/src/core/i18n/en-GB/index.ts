/* eslint-disable quote-props */
export default {
  'message': {
    'add': 'Add',
    'add-failure': 'Failed to add {0}',
    'add-success': 'Successfully added {0}',
    'add-text-snippet': 'Add text snippet',
    'app-name': 'Text Snippets',
    'cancel': 'Cancel',
    'confirm': 'Confirm',
    'copy-to-clipboard': 'Copy to clipboard',
    'copy-to-clipboard-failure': 'Text could not be copied to clipboard',
    'copy-to-clipboard-success': 'Text was copied to clipboard',
    'dark-theme': 'Dark theme',
    'date-time-added': 'Added on',
    'date-time-updated': 'Updated on',
    'de-DE': 'German',
    'de-DE-native': 'Deutsch',
    'delete': 'Delete',
    'delete-failure': 'Failed to delete {0}',
    'delete-message': 'Do you really want to delete {0}?',
    'delete-success': 'Successfully deleted {0}',
    'delete-title': 'Delete?',
    'description': 'Description | Descriptions',
    'en-GB': 'English',
    'en-GB-native': 'English',
    'es-ES': 'Spanish',
    'es-ES-native': 'Español',
    'home': 'Home | Homes',
    'id': 'Id | Ids',
    'invalid-length': 'Invalid length',
    'last-used': 'Last used',
    'light-theme': 'Light theme',
    'required': 'Required',
    'save': 'Save',
    'search': 'Search',
    'ta-LK': 'Tamil',
    'ta-LK-native': 'தமிழ்',
    'tag': 'Tag | Tags',
    'text-snippet': 'Text Snippet | Text Snippets',
    'update': 'Update',
    'update-failure': 'Failed to update {0}',
    'update-success': 'Successfully updated {0}',
    'update-text-snippet': 'Update text snippet',
    'value': 'Value | Values'
  }
};
