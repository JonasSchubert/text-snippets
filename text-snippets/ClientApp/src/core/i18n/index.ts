import Vue from 'vue';
import VueI18n from 'vue-i18n';

import enGb from './en-GB';

Vue.use(VueI18n);

export const defaultLocale = 'en-GB';
export const defaultMessages = { [defaultLocale]: enGb };

const i18n = new VueI18n({
  fallbackLocale: defaultLocale,
  locale: defaultLocale,
  messages: defaultMessages,
  silentTranslationWarn: process.env.NODE_ENV === 'production'
});

export default i18n;
