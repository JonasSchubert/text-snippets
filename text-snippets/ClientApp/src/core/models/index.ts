export * from './root-state.d';
