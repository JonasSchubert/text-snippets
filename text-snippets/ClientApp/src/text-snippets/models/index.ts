export * from './text-snippet.d';
export * from './text-snippets-state.d';
