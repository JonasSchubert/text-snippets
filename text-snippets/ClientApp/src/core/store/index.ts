import Vue from 'vue';
import Vuex, { Store, StoreOptions } from 'vuex';
import { createI18nModule } from '@/core/i18n/module-helper';
import { RootState } from '@/core/models';
import { ModuleType as LocalStorageModule } from "@/libs/local-storage/store/types";
import { createLocalStorageModule } from "@/libs/local-storage/store";
import { createModule as createTextSnippetsModule } from '@/text-snippets/store';
import { ModuleType as TextSnippetsModule } from '@/text-snippets/store/types';
import { createGetters } from './getters';
import { createMutations } from './mutations';
import { createState } from './state';

Vue.use(Vuex);

export const createStore = (vuetify: any): Store<RootState> => {
  const store: StoreOptions<RootState> = {
    getters: createGetters(),
    modules: {
      i18n: createI18nModule(),
      [LocalStorageModule]: createLocalStorageModule<RootState>('text-snippets', vuetify),
      [TextSnippetsModule]: createTextSnippetsModule()
    },
    mutations: createMutations(),
    state: createState(),
    strict: process.env.NODE_ENV !== 'production'
  };

  return new Store<RootState>(store);
};
