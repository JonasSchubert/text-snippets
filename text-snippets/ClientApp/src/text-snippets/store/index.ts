import { Module } from 'vuex';
import { RootState } from '@/core/models';
import { TextSnippetsState } from '@/text-snippets/models';
import { createActions } from './actions';
import { createGetters } from './getters';
import { createMutations } from './mutations';
import { createState } from './state';

export const createModule = (): Module<TextSnippetsState, RootState> => ({
  namespaced: true,
  actions: createActions(),
  getters: createGetters(),
  mutations: createMutations(),
  state: createState()
});
