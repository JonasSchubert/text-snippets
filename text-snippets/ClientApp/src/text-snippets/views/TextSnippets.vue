<template>
  <v-row class="fill-height" no-gutters>
    <v-col>
      <v-card height="100%">
        <v-toolbar extension-height="10" flat>
          <v-subheader>{{ $tc("message.text-snippet", list.length) }}</v-subheader>
          <v-spacer></v-spacer>
          <v-text-field
            :label="$t('message.search')"
            append-icon="mdi-magnify"
            dense
            hide-details
            single-line
            v-model="search"
          ></v-text-field>

          <template v-slot:extension>
            <v-tooltip bottom>
              <template v-slot:activator="{ on }">
                <v-fab-transition>
                  <v-btn
                    @click="addItem"
                    absolute
                    bottom
                    class="mr-1"
                    color="accent"
                    fab
                    right
                    small
                    v-on="on"
                  >
                    <v-icon>mdi-plus</v-icon>
                  </v-btn>
                </v-fab-transition>
              </template>
              <span>{{ $t("message.add") }}</span>
            </v-tooltip>
          </template>
        </v-toolbar>

        <v-data-table
          :footer-props="{
            'items-per-page-options': [5, 15, 25, 50]
          }"
          :headers="headers"
          :items="list"
          :items-per-page="25"
          :loading="isLoading"
          :search="search"
          :sort-by="['dateTimeAdded']"
          :sort-desc="[false]"
          dense
          multi-sort
        >
          <template v-slot:item.value="{ item }">
            {{ item.value }}
            <v-tooltip bottom>
              <template v-slot:activator="{ on }">
                <v-btn icon ripple @click="copyToClipboard(item.value)" v-on="on">
                  <v-icon>mdi-content-copy</v-icon>
                </v-btn>
              </template>
              <span>{{ $t("message.copy-to-clipboard") }}</span>
            </v-tooltip>
          </template>

          <template v-slot:item.actions="{ item }">
            <v-tooltip bottom>
              <template v-slot:activator="{ on }">
                <v-icon class="mr-2" small v-on="on" @click="editItem(item)">
                  mdi-pencil
                </v-icon>
              </template>
              <span>{{ $t("message.update") }}</span>
            </v-tooltip>

            <v-tooltip bottom>
              <template v-slot:activator="{ on }">
                <v-icon color="error" small v-on="on" @click="requestDelete(item)">mdi-delete</v-icon>
              </template>
              <span>{{ $t("message.delete") }}</span>
            </v-tooltip>
          </template>
        </v-data-table>

        <confirm-dialog ref="deleteConfirmDialog" @confirm="onDeleteConfirmed"></confirm-dialog>
        <edit-text-snippet-dialog ref="editTextSnippetDialog"></edit-text-snippet-dialog>
      </v-card>
    </v-col>
  </v-row>
</template>

<script lang="ts">
import Vue from "vue";
import { Component } from "vue-property-decorator";
import { mapGetters } from "vuex";
import { snackbar } from '@/libs/core/controls/app-snackbar/snackbar';
import { ConfirmDialogConfig } from "@/libs/core/dialogs/confirm-dialog";
import EditTextSnippetDialog from "@/text-snippets/dialogs/edit-text-snippet-dialog/edit-text-snippet-dialog.vue";
import { TextSnippet } from "@/text-snippets/models";
import { ActionTypes, GetterTypes, ModuleType } from "@/text-snippets/store/types";

@Component({
  name: "text-snippets",
  components: {
    "edit-text-snippet-dialog": EditTextSnippetDialog
  },
  computed: {
    ...mapGetters(ModuleType, [GetterTypes.isLoading, GetterTypes.list])
  }
})
export default class TextSnippets extends Vue {
  $refs!: {
    deleteConfirmDialog: any;
    editTextSnippetDialog: any;
  };

  public search = "";

  get headers(): any[] {
    return [
      { text: this.$tc("message.tag", 1), value: "tag", sortable: true },
      { text: this.$tc("message.value", 1), value: "value", sortable: true },
      { text: this.$tc("message.description", 1), value: "description", sortable: true },
      { text: this.$t("message.date-time-added"), value: "dateTimeAdded", sortable: true },
      { text: this.$t("message.date-time-updated"), value: "dateTimeUpdated", sortable: true },
      { text: "", value: "actions", sortable: false }
    ];
  }

  public addItem(): void {
    (this.$refs.editTextSnippetDialog as any).open();
  }

  // eslint-disable-next-line class-methods-use-this
  public copyToClipboard(value: string): void {
    navigator.clipboard
      .writeText(value)
      .then(() => snackbar.info({ message: "message.copy-to-clipboard-success" }))
      .catch(() => snackbar.warning({ message: "message.copy-to-clipboard-failure" }));
  }

  public editItem(item: TextSnippet): void {
    (this.$refs.editTextSnippetDialog as any).open(item);
  }

  public onDeleteConfirmed(item: TextSnippet): void {
    this.$store.dispatch(`${ModuleType}/${ActionTypes.delete}`, { item });
  }

  public requestDelete(item: TextSnippet): void {
    const config: ConfirmDialogConfig<TextSnippet> = {
      buttons: {
        cancel: {
          text: "message.cancel"
        },
        confirm: {
          text: "message.delete",
          textColor: "error"
        }
      },
      message: "message.delete-message",
      messageParams: [item.value.toString()],
      meta: item,
      title: "message.delete-title"
    };
    this.$refs.deleteConfirmDialog.open(config);
  }
}
</script>
