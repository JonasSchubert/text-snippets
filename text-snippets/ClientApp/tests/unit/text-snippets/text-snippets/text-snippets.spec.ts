import { shallowMount } from '@vue/test-utils';
import i18n from '@/core/i18n';
import vuetify from '@/core/plugins/vuetify.plugin';
import { createStore } from '@/core/store';
import { TextSnippet } from '@/text-snippets/models';
import TextSnippets from '@/text-snippets/views/TextSnippets.vue';

const store = createStore(vuetify);

// Mount the component
const wrapper = shallowMount(TextSnippets, {
  i18n,
  store,
  vuetify
});

describe('TextSnippets', () => {
  describe('computed', () => {
    describe('headers', () => {
      test('should not change unexpected', () => {
        // Arrange && Act
        const { headers } = wrapper.vm as any;

        // Assert
        expect(headers).toMatchSnapshot();
      });
    });
  });

  describe('methods', () => {
    describe('onDeleteConfirmed', () => {
      test('should call $store.dispatch', () => {
        // Arrange
        const item: TextSnippet = {
          id: 2,
          value: "42"
        } as unknown as TextSnippet;
        store.dispatch = jest.fn();

        // Act
        (wrapper.vm as any).onDeleteConfirmed(item);

        // Assert
        expect(store.dispatch).toHaveBeenCalledTimes(1);
        expect(store.dispatch).toHaveBeenCalledWith('text-snippets/delete', { item });
      });
    });

    describe('requestDelete', () => {
      test('should call $refs.deleteConfirmDialog.open', () => {
        // Arrange
        const item: TextSnippet = {
          id: 2,
          value: "42"
        } as unknown as TextSnippet;
        (wrapper.vm as any).$refs = {
          deleteConfirmDialog: { open: jest.fn() }
        };

        // Act
        (wrapper.vm as any).requestDelete(item);

        // Assert
        expect((wrapper.vm as any).$refs.deleteConfirmDialog.open).toHaveBeenCalledTimes(1);
        expect((wrapper.vm as any).$refs.deleteConfirmDialog.open).toHaveBeenCalledWith({
          buttons: {
            cancel: {
              text: "message.cancel"
            },
            confirm: {
              text: "message.delete",
              textColor: "error"
            }
          },
          message: "message.delete-message",
          messageParams: ["42"],
          meta: item,
          title: "message.delete-title"
        });
      });
    });
  });
});
