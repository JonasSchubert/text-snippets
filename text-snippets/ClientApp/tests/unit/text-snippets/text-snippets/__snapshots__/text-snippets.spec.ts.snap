// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextSnippets computed headers should not change unexpected 1`] = `
Array [
  Object {
    "sortable": true,
    "text": "Tag",
    "value": "tag",
  },
  Object {
    "sortable": true,
    "text": "Value",
    "value": "value",
  },
  Object {
    "sortable": true,
    "text": "Description",
    "value": "description",
  },
  Object {
    "sortable": true,
    "text": "Added on",
    "value": "dateTimeAdded",
  },
  Object {
    "sortable": true,
    "text": "Updated on",
    "value": "dateTimeUpdated",
  },
  Object {
    "sortable": false,
    "text": "",
    "value": "actions",
  },
]
`;
